extensions/resteasy-reactive/quarkus-rest-client-microprofile/deployment/src/main/java/io/quarkus/resteasy/reactive/client/microprofile/deployment/ReactiveResteasyMpClientProcessor.java

```diff
@@ -0,0 +1,289 @@
+    }
+
+    @BuildStep
+    void addMpClientEnricher(BuildProducer<JaxrsClientEnricherBuildItem> enrichers) {
+        enrichers.produce(new JaxrsClientEnricherBuildItem(new MicroProfileRestClientEnricher()));
+    }
+
+    private void searchForJaxRsMethods(List<MethodInfo> listOfKnownMethods, ClassInfo startingInterface, CompositeIndex index) {
+        for (MethodInfo method : startingInterface.methods()) {
+            if (isRestMethod(method)) {
+                listOfKnownMethods.add(method);
+            }
+        }
+        List<DotName> otherImplementedInterfaces = startingInterface.interfaceNames();
+        for (DotName otherInterface : otherImplementedInterfaces) {
+            ClassInfo superInterface = index.getClassByName(otherInterface);
+            if (superInterface != null)
+                searchForJaxRsMethods(listOfKnownMethods, superInterface, index);
+        }
+    }
+
+    @BuildStep
+    void registerHeaderFactoryBeans(CombinedIndexBuildItem index,
+            BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
+        Collection<AnnotationInstance> annotations = index.getIndex().getAnnotations(DotNames.REGISTER_CLIENT_HEADERS);
+
+        for (AnnotationInstance registerClientHeaders : annotations) {
+            AnnotationValue value = registerClientHeaders.value();
+            if (value != null) {
+                Type clientHeaderFactoryType = value.asClass();
+                String factoryTypeName = clientHeaderFactoryType.name().toString();
+                if (!MicroProfileRestClientEnricher.DEFAULT_HEADERS_FACTORY.equals(factoryTypeName)) {
+                    additionalBeans.produce(AdditionalBeanBuildItem.unremovableOf(factoryTypeName));
+                }
+            }
+        }
+    }
+
+    @BuildStep
+    void addRestClientBeans(Capabilities capabilities,
+            CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<GeneratedBeanBuildItem> generatedBeans) {
+
+        CompositeIndex index = CompositeIndex.create(combinedIndexBuildItem.getIndex());
+        Set<AnnotationInstance> registerRestClientAnnos = new HashSet<>(index.getAnnotations(REGISTER_REST_CLIENT));
+
+        for (AnnotationInstance registerRestClient : registerRestClientAnnos) {
+            ClassInfo jaxrsInterface = registerRestClient.target().asClass();
+            if (Modifier.isAbstract(jaxrsInterface.flags())) {
+
+                List<MethodInfo> restMethods = new ArrayList<>();
+
+                // search this class and its super interfaces for jaxrs methods
+                searchForJaxRsMethods(restMethods, jaxrsInterface, index);
+                if (restMethods.isEmpty()) {
+                    continue;
+                }
+
+                String wrapperClassName = jaxrsInterface.name().toString() + CDI_WRAPPER_SUFFIX;
+                try (ClassCreator classCreator = ClassCreator.builder()
```

Comment:
I find it very helpful to show in comments an example of the generated code

Reply:
+1

Reply:
done